src/gui/render_glsl.h

```diff
@@ -18,559 +18,569 @@
-"attribute vec4 a_position;\n"
-"void main() {\n"
-"  gl_Position = a_position;\n"
-"  v_texCoord = vec2(a_position.x+1.0,1.0-a_position.y)/2.0*rubyInputSize;\n"
-"}\n"
-""
-"#elif defined(FRAGMENT)\n"
-"uniform sampler2D rubyTexture;\n"
-""
-"void main() {\n"
-"  vec2 prescale = vec2(2.0);\n"
-"  vec2 texel = v_texCoord;\n"
-"  vec2 texel_floored = floor(texel);\n"
-"  vec2 s = fract(texel);\n"
-"  vec2 region_range = vec2(0.5) - vec2(0.5) / prescale;\n"
-""
-"  vec2 center_dist = s - 0.5;\n"
-"  vec2 f = (center_dist - clamp(center_dist, -region_range, region_range)) * prescale + vec2(0.5);\n"
-""
-"  vec2 mod_texel = min(texel_floored + f, rubyInputSize-0.5);\n"
-"  vec4 p = texture2D(rubyTexture, mod_texel/rubyTextureSize);\n"
-"  float ss = abs(s.y*2.0-1.0);\n"
-"  p -= p*ss*3.0/8.0;\n"
-""
-"  gl_FragColor = p;\n"
-"}\n"
-"#endif";
-
-static const char tv3x_glsl[] =
-"varying vec2 v_texCoord;\n"
-"uniform vec2 rubyInputSize;\n"
-"uniform vec2 rubyOutputSize;\n"
-"uniform vec2 rubyTextureSize;\n"
-""
-"#if defined(VERTEX)\n"
-"attribute vec4 a_position;\n"
-"void main() {\n"
-"  gl_Position = a_position;\n"
-"  v_texCoord = vec2(a_position.x+1.0,1.0-a_position.y)/2.0*rubyInputSize;\n"
-"}\n"
-""
-"#elif defined(FRAGMENT)\n"
-"uniform sampler2D rubyTexture;\n"
-""
-"void main() {\n"
-"  vec2 prescale = vec2(3.0);\n"
-"  vec2 texel = v_texCoord;\n"
-"  vec2 texel_floored = floor(texel);\n"
-"  vec2 s = fract(texel);\n"
-"  vec2 region_range = vec2(0.5) - vec2(0.5) / prescale;\n"
-""
-"  vec2 center_dist = s - 0.5;\n"
-"  vec2 f = (center_dist - clamp(center_dist, -region_range, region_range)) * prescale + vec2(0.5);\n"
-""
-"  vec2 mod_texel = min(texel_floored + f, rubyInputSize-0.5);\n"
-"  vec4 p = texture2D(rubyTexture, mod_texel/rubyTextureSize);\n"
-"  float ss = abs(s.y*2.0-1.0);\n"
-"  p -= p*ss*11.0/16.0;\n"
-""
-"  gl_FragColor = p;\n"
-"}\n"
-"#endif";
-
-static const char sharp_glsl[] =
-"varying vec2 v_texCoord;\n"
-"uniform vec2 rubyInputSize;\n"
-"uniform vec2 rubyOutputSize;\n"
-"uniform vec2 rubyTextureSize;\n"
-"varying vec2 prescale; // const set by vertex shader\n"
-""
-"#if defined(VERTEX)\n"
-"attribute vec4 a_position;\n"
-"void main() {\n"
-"  gl_Position = a_position;\n"
-"  v_texCoord = vec2(a_position.x+1.0,1.0-a_position.y)/2.0*rubyInputSize;\n"
-"  prescale = ceil(rubyOutputSize / rubyInputSize);\n"
-"}\n"
-""
-"#elif defined(FRAGMENT)\n"
-"uniform sampler2D rubyTexture;\n"
-""
-"void main() {\n"
-"  const vec2 halfp = vec2(0.5);\n"
-"  vec2 texel_floored = floor(v_texCoord);\n"
-"  vec2 s = fract(v_texCoord);\n"
-"  vec2 region_range = halfp - halfp / prescale;\n"
-""
-"  vec2 center_dist = s - halfp;\n"
-"  vec2 f = (center_dist - clamp(center_dist, -region_range, region_range)) * prescale + halfp;\n"
-""
-"  vec2 mod_texel = min(texel_floored + f, rubyInputSize-halfp);\n"
-"  gl_FragColor = texture2D(rubyTexture, mod_texel / rubyTextureSize);\n"
-"}\n"
-"#endif";
+static const char advinterp2x_glsl[] = R"GLSL(
```

Comment:
> R"GLSL(

At first, I wanted to suggest - hey, let's just use default raw string delimiter - but after thinking about this for a while - using `GLSL` in here is much better.

Thanks to this specific delimiter, we have several options for making clang format recognize content of the string to help us with autoformatting in the future (simplest way will be to adjust our script `scripts/format-commit.sh`). This is of course outside of scope of this PR.

What I wanted to say is: great job :+1: , thanks :)